src/Raven.Studio/typescript/viewmodels/database/indexes/editIndex.ts

```diff
@@ -270,22 +270,24 @@ class editIndex extends viewModelBase {
     }
 
     private fetchCustomAnalyzers() {
-        const db = this.activeDatabase();
-        
-        new getCustomAnalyzersCommand(db, true)
+        new getCustomAnalyzersCommand(this.activeDatabase(), true)
             .execute()
             .done((customAnalyzers) => {
-                const analyzerNames = customAnalyzers.map(x => x.Name);
-                this.editedIndex().customAnalyzers(analyzerNames);
-                
-                this.editedIndex().fields().forEach(x => x.addCustomAnalyzers(analyzerNames));
-                
-                const defaultFieldOptions = this.editedIndex().defaultFieldOptions();
-                if (defaultFieldOptions) {
-                    defaultFieldOptions.addCustomAnalyzers(analyzerNames);
-                }
+                this.registerCustomAnalyzers(customAnalyzers.map(x => x.Name))
             });
     }
+    
+    private registerCustomAnalyzers(analyzerNames: string[]) {
```

Comment:
I meant to move that inside this.editedIndex class ;-)

Reply:
(indexDefinition)

Reply:
done (forced pushed this)